### ISSUE TITLE
Deepcopy broken for MatthewsCorrCoef

### ISSUE THREAD

**Issue by hanlint** [Lightning-AI/metrics#1283]
## 🐛 Bug

The new release of `torchmetrics` breaks deepcopy for `MatthewsCorrCoef`

### To Reproduce
```
from torchmetrics import MatthewsCorrCoef
from copy import deepcopy
matthews_corrcoef = MatthewsCorrCoef(num_classes=2)
deepcopy(matthews_corrcoef)
```

Has the following error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/copy.py", line 264, in _reconstruct
    y = func(*args)
  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/copyreg.py", line 91, in __newobj__
    return cls.__new__(cls, *args)
TypeError: __new__() missing 1 required positional argument: 'num_classes'
```

This was _not_ an error in previous versions of `torchmetrics`. 

### Expected behavior

Deepcopy is successful

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source): `pip install torchmetrics==0.10.0`
- Python & PyTorch Version (e.g., 1.0): `Python 3.8.9 (default, Mar 30 2022, 13:51:16) `
- Any other relevant information such as OS (e.g., Linux): Max OS X

### Additional context

This bug is blocking our adoption of the newest torchmetrics release.

**Comment by stancld** [comment]
Hi @hanlint, this is a similar issue to #1274, which has been resolved in #1250 after #1249. I can confirm, everything works fine on `master`. Will keep you updated

**Comment by hanlint** [comment]
Great, thank you!
